src/query/ts/m3db/encoded_step_iterator.go

```diff
@@ -32,14 +30,7 @@ import (
 	"github.com/m3db/m3/src/query/ts/m3db/consolidators"
 )
 
-type peekValue struct {
-	started  bool
-	finished bool
-	point    ts.Datapoint
-}
-
 type encodedStepIter struct {
-	mu           sync.RWMutex
```

Comment:
Re: Removing the mutex, is there no concurrent access now?

Reply:
Yeah there wasn't really any before either, concurrent access on an iterator is a bit weird but was going to be necessary for a proposed lookback fix for split blocks.